torchvision/transforms.py

```diff
@@ -918,6 +914,42 @@ def __call__(self, img):
         return ten_crop(img, self.size, self.vertical_flip)
 
 
+class LinearTransformation(object):
+    """Transform a tensor image with a square transformation matrix computed
+    offline.
+
+    Given transformation_matrix, will flatten the torch.*Tensor, compute the dot
+    product with the transformation matrix and reshape the tensor to its
+    original shape.
+
+    Applications:
+    - whitening: zero-center the data, compute the data covariance matrix
+                 [D x D] with np.dot(X.T, X), perform SVD on this matrix and
+                 pass it as transformation_matrix.
+
+    Args:
+        transformation_matrix (Tensor): tensor [D x D], D = C x H x W
+    """
+
+    def __init__(self, transformation_matrix):
+        assert transformation_matrix.size(0) == transformation_matrix.size(1)
+        self.transformation_matrix = transformation_matrix
+
+    def __call__(self, tensor):
+        """
+        Args:
+            tensor (Tensor): Tensor image of size (C, H, W) to be whitened.
+
+        Returns:
+            Tensor: Transformed image.
+        """
+        assert tensor.size(0) * tensor.size(1) * tensor.size(2) == self.transformation_matrix.size(0)
```

Comment:
Same comment, could you add an error message?

Reply:
Ok, do I leave the assert or raise a ValueError here?

Reply:
Maybe raise `ValueError`, I think it's better.

Reply:
For this:

`ValueError: tensor and transformation matrix have incompatible shape. [%dim0 x %dim1 x %dim2] != %dim_transformation_matrix`

It would look like

`ValueError: tensor and transformation matrix have incompatible shape. [3 x 32 x 32] != 100`

for example. Do you think is fine?

I'm really not that good at writing error messages

Reply:
That's perfect, thanks!